### ISSUE TITLE
Upgrade grpc version to 1.0.4+

### ISSUE THREAD

**Issue by Random-Liu** [kubernetes/kubernetes#43040]
Ref #40156

We should update grpc version to 1.0.4+, or else we'll get the following build error when we vendor a `pb.go` built with grpc 1.0.4+.
```
# k8s.io/kubernetes/vendor/github.com/coreos/rkt/api/v1alpha
vendor/github.com/coreos/rkt/api/v1alpha/api.pb.go:844: undefined: grpc.SupportPackageIsVersion3
# k8s.io/kubernetes/vendor/github.com/coreos/etcd/etcdserver/etcdserverpb
vendor/github.com/coreos/etcd/etcdserver/etcdserverpb/rpc.pb.go:2082: undefined: grpc.SupportPackageIsVersion3
```

Actually, all our dependencies are using grpc 1.0.4+ now:
* etcd 1.0.4: https://github.com/coreos/etcd/blob/master/glide.yaml#L99
* rkt 1.0.4: https://github.com/coreos/rkt/blob/master/glide.yaml#L244

And other vendors are also using 1.0.4+:
* Docker grpc 1.0.4: https://github.com/docker/docker/blob/master/vendor.conf#L58
* Containerd grpc 1.0.5: https://github.com/docker/containerd/blob/master/vendor.conf#L23

Will the update break compatibility? @yifan-gu @hongchaodeng 

/cc @kubernetes/sig-api-machinery-misc @mikebrow

**Comment by Random-Liu** [comment]
Ref https://github.com/kubernetes/kubernetes/pull/41427

**Comment by timothysc** [comment]
All merged in #41427

**Comment by Random-Liu** [comment]
@timothysc Sooooooo quickly! Thanks a lot!
